### PR TITLE
Feature/drop 36/view onboarding medication reminders

### PR DESCRIPTION
**Pull request by CalebHasch**
## Describe your changes
I created addMoreSection component that duplicates the passed element and used this in the medication reminders route to dynamically add and remove the reminders and medications.

## Issue ticket number and link
ticket#: 36
link: https://tripleten-apiary.atlassian.net/jira/software/projects/DROP/boards/5?selectedIssue=DROP-36

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [x] I have made any necessary updates to Storybook to accompany these changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/components/form/add-more-section.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <>
+    {childList.map((child, i) => {
+      const index = i + 1;
+
+      return (
+        <div key={child}>
+          {React.isValidElement(children)
+          ? React.cloneElement(children, { index, onDelete: removeChild, elementId: child })
```
Comment (Aivaxela) [inline src/components/form/add-more-section.tsx:40]: nice use of React's cloneElement! this seems to be a good use-case for that function. the React docs do mention that this function can lead to "fragile code" but perhaps having the isValidElement check beforehand will help with this?

File: src/components/form/time-picker.tsx [hunk 2]
```diff
@@ -138,6 +140,14 @@ type Props<T extends FieldValues> = UseFormRegisterReturn<string> &
...
           <option value="PM">PM</option>
         </select>
+        <Button
+            type="button"
+            variant="icon"
+            buttonText=""
+            className="ml-auto"
+            icon={faTrashCan}
+            onClick={() => deleteTimePicker()}
+          ></Button>
```
Comment (Aivaxela) [inline src/components/form/time-picker.tsx:150]: I think the plan - according to Josh - was to try and keep the trash button out of the time-picker element, to make sure each individual component as simple (and reusable) as possible. Maybe there is some way the trash icon can be added to the add-more-section?

File: src/routes/onboarding/set-medication-reminders/medication-section.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+              variant="icon"
+              icon={faTrashCan}
+              className="ml-auto"
+              onClick={() => {
+                if (onDelete && elementId) {
+                  onDelete(elementId);
+                  unregister(inputName);
+                }
+              }}
+            />
```
Comment (Aivaxela) [inline src/routes/onboarding/set-medication-reminders/medication-section.tsx:37]: again I wonder if the button can be part of the add-more-section somehow, to ensure all the control for managing the generic list is contained?

File: src/routes/onboarding/set-medication-reminders/medication-time-input.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                  })}
+                  labelText={`Reminder ${index}`}
+                  hintText="Choose a time"
+                  setValue={(name, value) => field.onChange({ target: { name, value } })}
+                  feedback={errors.inputName?.message as string | undefined}
+                  delete={() => {
+                    if (onDelete && elementId) {
+                      onDelete(elementId);
+                      unregister(inputName);
+                    }
```
Comment (Aivaxela) [inline src/routes/onboarding/set-medication-reminders/medication-time-input.tsx:46]: same thing here, with the delete button

File: src/routes/onboarding/set-medication-reminders/medication-section.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+export const MedicationSection: React.FC<Props> = ({ index, onDelete, elementId}) =>  {
+
+  const {
+    register,
+    unregister,
+    formState: {errors},
+  } = useForm();
+
+  const inputName = `medication${elementId}` || "medication6"
```
Comment (Aivaxela) [inline src/routes/onboarding/set-medication-reminders/medication-section.tsx:21]: a bit confused about the fallback to "medication6" but maybe I am missing something 😅

File: src/routes/onboarding/set-medication-reminders/medication-reminder-form.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+export const MedicationReminderForm: React.FC = () => {
+  return (
+    <div className="flex flex-col bg-lightYellow max-w-screen-md pb-8 px-[10px] relative m-auto">
+      <h2 className="font-text text-section-header font-medium leader-[38.4px] text-center max-w-[358px] mx-auto">Set your medication reminders</h2>
+      <p className="font-text text-center max-w-[358px] mx-auto mt-4 mb-8">It’s important to be consistent in your medications, and notifications may help!</p>
+      <Form className="flex flex-col gap-5 ">
+        <Form.AddMoreSection buttonText="+ Add another medication">
+          <MedicationSection />
+        </Form.AddMoreSection>
```
Comment (Aivaxela) [inline src/routes/onboarding/set-medication-reminders/medication-reminder-form.tsx:13]: the implementation looks very clean!

**Review by Aivaxela — COMMENTED** [review]
Overall I think this is a very clean implementation! I left a couple of notes to maybe consider?
Also I'd be curious to see how well we'd actually be able to validate and submit all the form data here. I see you are using the useForm hook, so hopefully we can find a way to send those form values back to the submit function and on to the backend.

**Review by Brian-J-Wang — APPROVED** [review]
Looks good to me.

File: src/components/form/list-time-input.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+              </>
+            )}
+          />
+           <Button
+              variant="icon"
+              icon={faTrashCan}
+              className="absolute bottom-[22px] right-5"
+              onClick={() => {
+                if (onDelete && elementId) {
+                  console.log(inputName)
```
Comment (Aivaxela) [inline src/components/form/list-time-input.tsx:64]: leftover console.log statement? - probably needs deleted before moving to main

File: src/components/form/list-time-input.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+                />
+                
+              </>
+            )}
+          />
+           <Button
+              variant="icon"
+              icon={faTrashCan}
+              className="absolute bottom-[22px] right-5"
+              onClick={() => {
```
Comment (Aivaxela) [inline src/components/form/list-time-input.tsx:62]: might be a good idea to ensure the item isn't deleted if there is only 1 rendered (so the list is never empty) (if the user doesn't want reminders they'll just hit the skip button I believe)

File: src/routes/onboarding/set-medication-reminders/medication-section.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    register,
+    unregister,
+    formState: {errors},
+  } = useForm<TextInputValue>();
+
+
+  return (
+    <div className="flex flex-col gap-5 bg-white mx-auto min-w-[315px] max-w-[400px] px-4 py-3 w-full rounded-[16px]">
+          <div className="flex items-centerw">
+            <h3 className="font-text text-section-subtext font-normal leading-[24px] text-center">Medication {index}</h3>
```
Comment (Aivaxela) [inline src/routes/onboarding/set-medication-reminders/medication-section.tsx:27]: looking at the Figma, this h3 should have a font-weight of 500 (tailwind class is `font-normal`)
also, the font size should be 24px (tailwind `text-2xl`)

File: src/routes/onboarding/set-medication-reminders/medication-reminder-form.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { Form } from "../../../components"
+import { MedicationSection } from "./medication-section";
+
+export const MedicationReminderForm: React.FC = () => {
+  return (
+    <div className="flex flex-col bg-lightYellow max-w-screen-md pb-8 px-[10px] relative m-auto">
+      <h2 className="font-text text-section-header font-medium leader-[38.4px] text-center max-w-[358px] mx-auto">Set your medication reminders</h2>
```
Comment (Aivaxela) [inline src/routes/onboarding/set-medication-reminders/medication-reminder-form.tsx:8]: small Tailwind class typo here - the line-height class should be `leading-[38.4px]` not `leader`

**Review by Aivaxela — CHANGES_REQUESTED** [review]
Nice work with the add-more section! I was able to integrate my step into it pretty easily. 
I left a few comments on some very minor, simple things to fix.

File: src/routes/onboarding/set-medication-reminders/medication-reminder-form.tsx [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+      <p className="font-text text-center max-w-[358px] mx-auto mt-4 mb-8">It’s important to be consistent in your medications, and notifications may help!</p>
+      <Form className="flex flex-col gap-5 ">
+        <Form.AddMoreSection buttonText="+ Add another medication">
```
Comment (Brian-J-Wang) [inline src/routes/onboarding/set-medication-reminders/medication-reminder-form.tsx:11]: There is too much spacing between this button and the bottom medication section. The Figma doc says that there should be 20px between the two elements.

Your implementation
image

Figma Docs
image
image

**Review by Brian-J-Wang — CHANGES_REQUESTED** [review]
Just a small detail that I noticed while reviewing Riley's PR.

**Review by Brian-J-Wang — APPROVED** [review]
Looks good to me.

**Review by Aivaxela — APPROVED** [review]

**Review by jahorwitz — APPROVED** [review]

**Pull request merged**